src/Symfony/Component/Security/Core/Validator/Constraints/UserPasswordValidator.php

```diff
@@ -39,6 +39,10 @@ public function validate($password, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\UserPassword');
         }
 
+        if (null === $password || '' === $password) {
+            return;
```

Comment:
Now using **FOSUserBundle**'s **ChangePasswordFormType**, logged in users can change their own password without typing current password. I created a pull request:
[https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2582](https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2582)